### PR TITLE
Use s3 for processor assets

### PR DESCRIPTION
**Pull request by kasbah**
- Removed the http endpoint from processor completely
- Added Minio in development and actual S3 for staging
- For staging servers we can use the aws cli from the server itself to remove the bucket if we need to clear it out. 
- Switched processor testing to vitest with mocks for all dependent services except redis. 
    - So we have integration that have to be run with docker-compose.test-processor.yml with just the processor and redis docker containers
    - Made the readme test a unit test, this could be run without docker but it seems easier to just run it at the same time as the integration test.
    - Moved the test for finding header with empty cells (sangaboard-v0.3) to a npm-1-click-bom unit test

### CODE REVIEW

File: processor/src/tasks/processGerbers/index.ts [hunk 1]
```diff
@@ -236,19 +294,25 @@ function generateGerberInfo(zipPath, stackup, inputFiles, gerberInfoPath) {
   gerberInfo.width = Math.ceil(gerberInfo.width)
   gerberInfo.height = Math.ceil(gerberInfo.height)
 
-  return writeFile(path.join(gerberInfoPath), JSON.stringify(gerberInfo))
+  s3.uploadFileContents(
```
Comment (kasbah) [inline processor/src/tasks/processGerbers/index.ts:297]:
```suggestion
  return s3.uploadFileContents(
```

**Comment by AbdulrhmnGhanem** [comment]
The processor throws an error on startup
```
kitspace-processor-1    | /app/node_modules/@aws-sdk/smithy-client/dist-cjs/default-error-handler.js:8
kitspace-processor-1    |     const response = new exceptionCtor({
kitspace-processor-1    |                      ^
kitspace-processor-1    | 
kitspace-processor-1    | 502: UnknownError
kitspace-processor-1    |     at throwDefaultError (/app/node_modules/@aws-sdk/smithy-client/dist-cjs/default-error-handler.js:8:22)
kitspace-processor-1    |     at deserializeAws_restXmlCreateBucketCommandError (/app/node_modules/@aws-sdk/client-s3/dist-cjs/protocols/Aws_restXml.js:3165:51)
kitspace-processor-1    |     at processTicksAndRejections (node:internal/process/task_queues:96:5)
kitspace-processor-1    |     at async /app/node_modules/@aws-sdk/middleware-serde/dist-cjs/deserializerMiddleware.js:7:24
kitspace-processor-1    |     at async /app/node_modules/@aws-sdk/middleware-signing/dist-cjs/middleware.js:14:20
kitspace-processor-1    |     at async StandardRetryStrategy.retry (/app/node_modules/@aws-sdk/middleware-retry/dist-cjs/StandardRetryStrategy.js:51:46)
kitspace-processor-1    |     at async /app/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:6:22
kitspace-processor-1    |     at async file:///app/dist/src/s3.js:25:5 {
kitspace-processor-1    |   '$fault': 'client',
kitspace-processor-1    |   '$metadata': {
kitspace-processor-1    |     httpStatusCode: 502,
kitspace-processor-1    |     requestId: undefined,
kitspace-processor-1    |     extendedRequestId: undefined,
kitspace-processor-1    |     cfId: undefined,
kitspace-processor-1    |     attempts: 3,
kitspace-processor-1    |     totalRetryDelay: 455
kitspace-processor-1    |   },
kitspace-processor-1    |   head: { title: '502 Bad Gateway' },
kitspace-processor-1    |   body: {
kitspace-processor-1    |     center: { h1: '502 Bad Gateway' },
kitspace-processor-1    |     hr: { center: 'nginx/1.17.10' }
kitspace-processor-1    |   }
kitspace-processor-1    | }
```

**Comment by AbdulrhmnGhanem** [comment]
When I restart the processor container it gets stuck at
```
kitspace-processor-1  | yarn install && yarn dev
kitspace-processor-1  | yarn install v1.22.19
kitspace-processor-1  | [1/5] Validating package.json...
kitspace-processor-1  | [2/5] Resolving packages...
kitspace-processor-1  | success Already up-to-date.
kitspace-processor-1  | Done in 0.43s.
kitspace-processor-1  | yarn run v1.22.19
kitspace-processor-1  | $ tsc-watch --noClear --onSuccess 'bash -c "yarn cp-assets && node ./dist/src/main.js"'
kitspace-processor-1  | 
kitspace-processor-1  | 
kitspace-processor-1  | ----------------------
kitspace-processor-1  | 9:23:57 AM - Starting compilation in watch mode...
kitspace-processor-1  | 
kitspace-processor-1  | 
kitspace-processor-1  | 9:24:01 AM - Found 0 errors. Watching for file changes.
kitspace-processor-1  | $ find ./src ! -name '*.ts' -exec cp -r '{}' ./dist/src/ \;
```

**Comment by AbdulrhmnGhanem** [comment]
> The processor throws an error on startup
 
My bad I should have rebuilt nginix, it works after rebuilding.

> When I restart the processor container it gets stuck at

It isn't actually stuck. But we used to have a message on startup, I think the old behavior makes more sense.

**Review by AbdulrhmnGhanem — APPROVED** [review]

**Comment by AbdulrhmnGhanem** [comment]
Shouldn't this part get updated for other staging servers too? or removed if it's handled by ansible?

https://github.com/kitspace/kitspace-v2/blob/015713fb691ff035104bc323d9983ba2118cd27a/.github/workflows/docker.yml#L224-L230

**Comment by kasbah** [comment]
> Shouldn't this part get updated for other staging servers too? or removed if it's handled by ansible?

Ansible just handles the aws cli. This handles passing the secrets to docker-compose. I updated this on master for the other branches because I thought I needed to do it there for the review deploy (I forgot the `review` action is run `on: push` and not `on: pull_request_target`). 

https://github.com/kitspace/kitspace-v2/blob/b496a6b3eb48af626a4d02cfd7201f96df9e87f8/.github/workflows/docker.yml#L189-L208

**Pull request merged**